### PR TITLE
feat: add footer

### PR DESCRIPTION
**Pull request by amt-augustineopokukesse**
added footer

### CODE REVIEW

File: apps/iglidur-designer/src/app/libs/header/src/lib/toolbar/toolbar.component.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@ import { Subject, takeUntil } from 'rxjs';
 
 @Component({
   selector: 'lib-toolbar',
+  encapsulation: ViewEncapsulation.None,
```
Comment (Bilal-Abubakari) [inline apps/iglidur-designer/src/app/libs/header/src/lib/toolbar/toolbar.component.ts:12]: Why this?

File: apps/iglidur-designer/src/app/pages/model/model.component.html [hunk 2]
```diff
@@ -39,10 +39,29 @@ <h2>Select Model</h2>
     />
     } }
 
-    <div id="model" [class.hidden]="screenshotUrl">
+    <!-- <div id="model" [class.hidden]="screenshotUrl"> -->
```
Comment (Bilal-Abubakari) [inline apps/iglidur-designer/src/app/pages/model/model.component.html:42]: Remove comment

File: apps/iglidur-designer/src/app/pages/model/model.component.html [hunk 3]
```diff
@@ -39,10 +39,29 @@ <h2>Select Model</h2>
...
       @if (modelUrl && !screenshotUrl) {
       <stl-model-viewer [stlModels]="[modelUrl]"></stl-model-viewer>
       }
     </div>
   </div>
+  <div class="footer">
+    <p class="footer-text">Please rate this configurator</p>
+    <div class="rating">
+      <mat-icon
+        *ngFor="let star of stars; let i = index"
```
Comment (Bilal-Abubakari) [inline apps/iglidur-designer/src/app/pages/model/model.component.html:53]: prefer new control flow

In fact add it to your eslint

File: apps/iglidur-designer/src/app/pages/model/model.component.html [hunk 4]
```diff
@@ -39,10 +39,29 @@ <h2>Select Model</h2>
...
+        (click)="rate(i + 1)"
+        [ngClass]="{ 'filled': i < rating }"
```
Comment (Bilal-Abubakari) [inline apps/iglidur-designer/src/app/pages/model/model.component.html:55]:
```suggestion
        [class.filled]="i < rating"
```

File: apps/iglidur-designer/src/app/pages/model/model.component.ts [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import { Component, OnDestroy, OnInit } from '@angular/core';
+import { AfterViewInit, Component, ElementRef, OnDestroy, OnInit, ViewChild, ViewEncapsulation } from '@angular/core';
```
Comment (Bilal-Abubakari) [inline apps/iglidur-designer/src/app/pages/model/model.component.ts:1]: This is not what the feature is supposed to be doing

**Review by Bilal-Abubakari — COMMENTED** [review]
If it is a footer feature then it should sole be about footer
But you have so many other stuffs inside this PR

File: apps/iglidur-designer/src/app/libs/header/custom-stl-viewer/custom-stl-viewer.component.html [hunk 6]
```diff
@@ -0,0 +1 @@
+<p>custom-stl-viewer works!</p>
```
[the remark this answers is not in the document]
Comment (amt-augustineopokukesse) [reply apps/iglidur-designer/src/app/libs/header/custom-stl-viewer/custom-stl-viewer.component.html:1]: I added it to test the navigation tabs

File: apps/iglidur-designer/src/app/libs/header/src/lib/toolbar/toolbar.component.ts (hunk 1 above)
Comment (amt-augustineopokukesse) [reply apps/iglidur-designer/src/app/libs/header/src/lib/toolbar/toolbar.component.ts:12]: It's been resolved.. I added it while trying to fix an angular material custom theme issue.

File: apps/iglidur-designer/src/app/pages/model/model.component.html (hunk 3 above)
Comment (amt-augustineopokukesse) [reply apps/iglidur-designer/src/app/pages/model/model.component.html:53]: I have added the rule to the eslint

File: apps/iglidur-designer/src/app/pages/model/model.component.ts (hunk 5 above)
Comment (amt-augustineopokukesse) [reply apps/iglidur-designer/src/app/pages/model/model.component.ts:1]: I had already implemented a model viewer so it formed part of the files for the PR.
This was before you told me to focus on the other features.

File: apps/iglidur-designer/src/app/libs/header/custom-stl-viewer/custom-stl-viewer.component.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (Bilal-Abubakari) [reply apps/iglidur-designer/src/app/libs/header/custom-stl-viewer/custom-stl-viewer.component.html:1]: Is this part of the fotter?

File: apps/iglidur-designer/src/app/libs/header/custom-stl-viewer/custom-stl-viewer.component.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
+// import { Component, ElementRef, Input, OnInit, ViewChild } from '@angular/core';
```
Comment (Bilal-Abubakari) [inline apps/iglidur-designer/src/app/libs/header/custom-stl-viewer/custom-stl-viewer.component.ts:1]: Commented code?

File: apps/iglidur-designer/src/app/libs/header/src/lib/toolbar/toolbar.component.html [hunk 8]
```diff
@@ -1,10 +1,20 @@
 <div class="toolbar">
-  <nav mat-tab-nav-bar class="mat-tab-header">
+  <nav mat-tab-nav-bar color="primary" class="mat-tab-header" [tabPanel]="tabPanel">
```
Comment (Bilal-Abubakari) [inline apps/iglidur-designer/src/app/libs/header/src/lib/toolbar/toolbar.component.html:2]: Has this got anything to do with footer?

File: apps/iglidur-designer/src/app/pages/model/model.component.html [hunk 9]
```diff
@@ -39,10 +38,31 @@ <h2>Select Model</h2>
...
     } }
 
-    <div id="model" [class.hidden]="screenshotUrl">
+    <div #rendererContainer [class.hidden]="screenshotUrl">
       @if (modelUrl && !screenshotUrl) {
       <stl-model-viewer [stlModels]="[modelUrl]"></stl-model-viewer>
       }
     </div>
   </div>
+  <div class="footer">
```
Comment (Bilal-Abubakari) [inline apps/iglidur-designer/src/app/pages/model/model.component.html:47]: Why is footer and model in the same component?

File: apps/iglidur-designer/src/app/pages/model/model.component.ts (hunk 5 above)
Comment (Bilal-Abubakari) [reply apps/iglidur-designer/src/app/pages/model/model.component.ts:1]: Do not mix features in one PR. You can always commit files that are solely about what you are doing

File: apps/iglidur-designer/src/custom-theme.scss [hunk 10]
```diff
@@ -9,24 +9,63 @@
...
+  // density: 0,
 ));
 
 // Include theme styles for core and each component used in your app.
 // Alternatively, you can import and @include the theme mixins for each component
 // that you are using.
-:root {
-  @include mat.all-component-themes($iglidur-designer-theme);
-}
+// :root {
```
Comment (Bilal-Abubakari) [inline apps/iglidur-designer/src/custom-theme.scss:66]: Commented code?

File: apps/iglidur-designer/src/custom-theme.scss [hunk 11]
```diff
@@ -9,24 +9,63 @@
...
-    tertiary: mat.$blue-palette,
+    // theme-type: light,
+    primary: $iglidur-primary,
+    accent: $iglidur-accent,
+    warn: $iglidur-warn,
   ),
-  density: (
-    scale: 0,
-  )
+  // typography: mat.define-typography-config(),
```
Comment (Bilal-Abubakari) [inline apps/iglidur-designer/src/custom-theme.scss:59]: Commented code?

**Review by Bilal-Abubakari — COMMENTED** [review]
This PR is about footer so I expect this Pr solely focus on footer not other stuffs.

File: apps/iglidur-designer/src/app/libs/header/custom-stl-viewer/custom-stl-viewer.component.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (amt-augustineopokukesse) [reply apps/iglidur-designer/src/app/libs/header/custom-stl-viewer/custom-stl-viewer.component.html:1]: No, it's not but it is a feature that was part of an earlier PR

**Review by Bilal-Abubakari — APPROVED** [review]
I don't like the idea of merging the wrong things or PRs into main

**Pull request merged**